src/library_tty.js

```diff
@@ -95,13 +95,31 @@ mergeInto(LibraryManager.library, {
         if (!tty.input.length) {
           var result = null;
           if (ENVIRONMENT_IS_NODE) {
-            result = process['stdin']['read']();
-            if (!result) {
-              if (process['stdin']['_readableState'] && process['stdin']['_readableState']['ended']) {
-                return null;  // EOF
+            // we will read data by chunks of BUFSIZE
+            var BUFSIZE = 256;
```

Comment:
Chunking is the way node's readSync works, in a low-level BYOB way, reminiscent of C's stdio.

The requirement placed on this method is simply to produce a byte from stdin. The extra data is stored in tty.input. If more data needs to be read, this gets run again, appending more data. 256 is an arbitrary figure, a power of 2 for optimal allocation, and big enough to avoid having to perform blocking I/O too often for common use-cases.